skimage/restoration/inpaint.py

```diff
@@ -121,16 +123,27 @@ def inpaint_biharmonic(img, mask, multichannel=False):
     img = skimage.img_as_float(img)
     mask = mask.astype(np.bool)
 
+    # Split inpainting mask into independent regions
+    kernel = ndi.morphology.generate_binary_structure(mask.ndim, 1)
+    kernel = ndi.iterate_structure(kernel, 2)
+    mask_dilated = ndi.morphology.binary_dilation(mask, structure=kernel)
```

Comment:
Won't this combine independent, adjacent sub-regions? Is it reasonable to just use the next two lines instead, labeling the actual mask instead of a dilated mask?

Asking mainly because binary dilation takes time, and operates on the entire image. I suspect that on big images with small-ish masks this step may eliminate the performance gains.